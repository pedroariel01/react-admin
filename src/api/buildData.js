function buildConnectorKanban(
  connectors,
  status_connector,
  show_client_name = false
) {
  let columns = {};
  let lands_result = { lanes: [] };

  for (var i = 0; i < status_connector.length; i++) {
    let status = status_connector[i];
    let new_column = {
      id: status.pk.toString(),
      title: status.name,
      style: {
        width: 280
      },
      cards: []
    };

    if (!(new_column.id in columns)) {
      columns[new_column.id] = new_column;
    }

    lands_result.lanes.push(new_column);
  }

  for (var i = 0; i < connectors.length; i++) {
    let connector = connectors[i];
    let status_connector = connector.status;
    let info_connector = "";

    if (show_client_name) {
      info_connector = `<i>${connector.manager.client.name}</i>`;
    }

    let description = `${connector.state} ${connector.county} ${connector.area}`;
    columns[status_connector.pk].cards.push({
      title: info_connector,
      description: description,
      id: connector.pk
    });
  }

  return lands_result;
}

export default buildConnectorKanban;
