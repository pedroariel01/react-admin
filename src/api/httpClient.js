import axios from "axios";

/**
 * Axios common configuration
 */
const config = {};

const configAssessor = {};

//Create axios with initial config
const apiClient = axios.create(config);

const apiClientAssessor = axios.create(configAssessor);

// const serverAresAddr = 'https://192.81.135.92:8000';

// @permission_classes((IsAuthenticated, ))

const authInterceptor = config => {
  const token = localStorage.getItem("jwtToken");
  // config.url  = serverAresAddr + config.url;
  // console.log(config.headers);
  config.headers.Authorization = "JWT " + token;
  return config;
};

const authAssessorInterceptor = configAssessor => {
  configAssessor.headers.Authorization = 'Bearer 3772bbcd739e97f17a7fb4b0c39ba3a97b531ec5';
  return configAssessor;
};

apiClientAssessor.interceptors.request.use(authAssessorInterceptor);

apiClient.interceptors.request.use(authInterceptor);


apiClient.interceptors.response.use(
  function(response) {
    // Do something with response data
    return response;
  },
  function(error) {
    console.log(error);
    if (error.response && error.response.status === 401) {
      localStorage.removeItem("jwtToken");
      window.location.href = "/admin/login";
    }
    return Promise.reject(error);
  }
);



apiClientAssessor.interceptors.response.use(
    function(response) {
    // Do something with response data
    return response;
  },
  function(error) {
    console.log(error);
    if (error.response.status === 401) {
      localStorage.removeItem("jwtToken");
      window.location.href = "/admin/login";
    }
    return Promise.reject(error);
  }

);

const serverAddr = 'http://45.33.62.162:8002';

// const serverAddr = 'http://127.0.0.1:8000';

export { apiClient, apiClientAssessor,serverAddr};
