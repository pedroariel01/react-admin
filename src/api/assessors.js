import { apiClient,serverAddr } from "./httpClient";



function getAssessor(connector_id) {
  return apiClient.get(`/connectors/connector/${connector_id}`);
}

function getAssessors(client_id) {
  if (client_id) {
    return apiClient({
      url: `/connectors/all?client_id=${client_id}`,
      method: "GET"
    });
  } else {
    return apiClient({
      url: `/connectors/all`,
      method: "GET"
    });
  }
}

function getAssociationClient(client_id) {
  if (client_id) {
    return apiClient({
      url: `/connectors/client/association?search=${client_id}`,
      method: "GET"
    });
  }

  return apiClient({
    url: "/connectors/client/association",
    method: "GET"
  });
}

function buildAssociation(client_id, connector_id, filter) {
  return apiClient.post("/connectors/association/connector", {
    client_id,
    connector_id,
    filter
  });
}

function filterConnectors() {}

function getStatusConnector() {}

function updateStatusConnector() {}

export {
  getAssessor,
  filterConnectors,
  getStatusConnector,
  updateStatusConnector,
  getAssessors,
  buildAssociation,
  getAssociationClient
};
