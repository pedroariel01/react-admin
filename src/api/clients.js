import { apiClient } from "./httpClient";

function clients(page=0) {
  let pageNumb = 1;
  if(page){
    const aux  = page.match('page=([0-9]+)')
    if(aux){
      pageNumb=aux[1]
    }
  }
  return apiClient.get(`/connectors/clients?page=${pageNumb}`).then(res => {
    return res.data;
  });
}

function newClient(dataClient) {
  const data = {
    name: dataClient.name.value,
    description: dataClient.description.value,
    confluence_pages_urls: dataClient.confluence_pages_urls.value
  };

  return apiClient.post("/connectors/clients", data).then(res => {
    return res;
  });
}

function getClient(id) {
  return apiClient.get(`/connectors/client/${id}`).then(res => {
    return res;
  });
}

function updateClient(id, client) {
  return apiClient.put(`/connectors/client/${id}`, client).then(res => {
    return res;
  });
}

function deleteClient(client) {
  const { id } = client;
  return apiClient.delete(`/connectors/client/${id}`);
}

function getConnectors(client_id) {
  return apiClient({
    url: `/connectors/client/connector`,
    method: "GET",
    params: { search: client_id }
  });
}

function getClientNote(client_id) {
  return apiClient.get("/connectors/notes", { params: { search: client_id } });
}

function newClientNote(client_id, note) {
  return apiClient.post("/connectors/notes", { client: client_id, note });
}

// DEPTH_PRIORITY,
//       AUTOTHROTTLE_TARGET_CONCURRENCY,
//       AUTOTHROTTLE_START_DELAY

function runSpider(
  data_origin,
  data_final_schema,
  config_name,
  column_origin,
  log_name,
  concurrent_request,
  download_delay,
  enabledProxy,
  AUTOTHROTTLE_ENABLED,
  RANDOM_UA_PER_PROXY,
  AUTOTHROTTLE_DEBUG,
  ROBOTSTXT_OBEY,
  PROXYMESH_TIMEOUT,
  DEPTH_PRIORITY,
  AUTOTHROTTLE_TARGET_CONCURRENCY,
  AUTOTHROTTLE_START_DELAY,
  middlewareList
) {
  return apiClient.get("/connectors/run/spider", {
    params: {
      data_origin,
      data_final_schema,
      config_name,
      column_origin,
      log_name,
      concurrent_request,
      download_delay,
      enabledProxy,
      AUTOTHROTTLE_ENABLED,
      RANDOM_UA_PER_PROXY,
      AUTOTHROTTLE_DEBUG,
      ROBOTSTXT_OBEY,
      PROXYMESH_TIMEOUT,
      DEPTH_PRIORITY,
      AUTOTHROTTLE_TARGET_CONCURRENCY,
      AUTOTHROTTLE_START_DELAY,
      middlewareList
    }
  });
}

function getConfigs(name) {
  return apiClient.get("/connectors/configs", {
    params: { name }
  });
}

export {
  clients,
  newClient,
  getConnectors,
  deleteClient,
  updateClient,
  getClient,
  getClientNote,
  newClientNote,
  runSpider,
  getConfigs
};
