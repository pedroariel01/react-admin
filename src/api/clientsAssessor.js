import { apiClientAssessor,serverAddr } from "./httpClient";

function clientsAssessor() {
  return apiClientAssessor.get(`${serverAddr}/assessor/clients`).then(res => {
      // console.log(res);
    return res.data;
  });
}


function getClient(id) {
  return apiClientAssessor.get(`${serverAddr}/assessor/clients/${id}`).then(res => {

      return res.data;
  });
}

function getAllClientData(id) {

   var client_data = {};
   const getAllData = async() =>{
       await apiClientAssessor.get(`${serverAddr}/assessor/clients/${id}`).then(res => {
           client_data['client'] = res.data;
       });

       await apiClientAssessor.get(`${serverAddr}/assessor/clients/${id}/schemas`).then(res => {
          client_data['schemas']= res.data;
       });

       await apiClientAssessor.get(`${serverAddr}/assessor/clients/${id}/data_origins`).then(res => {
              client_data['data_origins']= res.data;

           });
       await apiClientAssessor.get(`${serverAddr}/assessor/clients/${id}/tasks`).then(res => {
                  client_data['tasks']= res.data;

               });
       return client_data;

   };
   return getAllData();

}



function newAssessorClient(name,country,state,county,config,is_scraping){

  const data =
      {
        name:name,
        country:country,
        state:state,
        county:county,
        assessor_config:config,
          scraping:is_scraping
      };
  return apiClientAssessor.post(`${serverAddr}/assessor/clients/`, data).then(res => {
    return res;
  });
}

function getNextRunList(){

    return apiClientAssessor.get(`${serverAddr}/assessor/assessor_task/next_running/`).then(res => {
      // console.log(res);
    return res.data;
  });
}

function getSchemas(state='', county=''){

        let data =
            {
                country: 'us'
            };
        if(state){
            data.state=state;
        }
        if(county){
            data.county = county
        }

        return apiClientAssessor.get(`${serverAddr}/assessor/schemas`).then(
            res=>{
                return res.data}
        )
}


function newConsolidationSchema(data_origin,data_origin_field,schema_1,schema_2,schema_name){
    const data= {
            schemas_to_merge:[schema_1,schema_2],
            schema_result:schema_name,
        data_origin:data_origin,
        data_origin_field:data_origin_field
    };
    return apiClientAssessor.post(`${serverAddr}/admin/consolidation`, data).then(res => {
    return res;
  });
}

function newAssessorTask(name,interval,client,enab,oneOff,task_type){
    const data = {
        name:name,
        interval:interval,
        client:client,
        enabled:enab,
        one_off:oneOff,
        task:task_type

    };
    return apiClientAssessor.post(`${serverAddr}/assessor/assessor_task/`, data).then(res => {

        return res;
      });

}

function newAssessorDataOrigin(name,field,client,layer,extract) {
    const data = {
        name:name,
        field:field,
        layer:layer,
        client:client,
        extract:extract
    };
    // console.log(data);

    return apiClientAssessor.post(`${serverAddr}/assessor/data_origin/`, data).then(res => {

        return res;
      });

}

function getIntervals(){
    return apiClientAssessor.get(`${serverAddr}/assessor/schedule`).then(res => {
    return res;
  });
}

function getTaskType(){
    return apiClientAssessor.get(`${serverAddr}/assessor/assessor_task/tasks/`).then(res => {
    return res.data;
  });
}

function getTaskResult(){
     return apiClientAssessor.get(`${serverAddr}/assessor/task_result_ext/root/`).then(res => {
    return res.data.results;
  });
}

async function   getTaskResultData(id){
    return await apiClientAssessor.get(`${serverAddr}/assessor/task_result_ext/${id}`).then(res=>{
        return res;
    })
}

function getTaskResultDataFamily(id){
    return apiClientAssessor.get(`${serverAddr}/assessor/task_result_ext/${id}/family`).then(res => {
    return res;
  });
}

function getSchema(id){
    return apiClientAssessor.get(`${serverAddr}/assessor/schemas/${id}`).then(res => {
    return res;
  });
}

function getTasks(){
    return apiClientAssessor.get(`${serverAddr}/assessor/assessor_task`).then(res => {
    return res;
  });
}



function updateClient(id, client) {
  return apiClientAssessor.put(`/connectors/client/${id}`, client).then(res => {
    return res;
  });
}

function deleteClient(client) {
  const { id } = client;
  return apiClientAssessor.delete(`/connectors/client/${id}`);
}




function getDataOrigins(){
  return apiClientAssessor.get(`${serverAddr}/assessor/data_origin`).then(res => {
    return res.data;
  });
}
// DEPTH_PRIORITY,
//       AUTOTHROTTLE_TARGET_CONCURRENCY,
//       AUTOTHROTTLE_START_DELAY

function runSpider(
  data_origin,
  data_final_schema,
  config_name,
  column_origin,
  log_name,
  concurrent_request,
  download_delay,
  enabledProxy,
  AUTOTHROTTLE_ENABLED,
  RANDOM_UA_PER_PROXY,
  AUTOTHROTTLE_DEBUG,
  ROBOTSTXT_OBEY,
  PROXYMESH_TIMEOUT,
  DEPTH_PRIORITY,
  AUTOTHROTTLE_TARGET_CONCURRENCY,
  AUTOTHROTTLE_START_DELAY,
  middlewareList
) {
  return apiClientAssessor.get("/connectors/run/spider", {
    params: {
      data_origin,
      data_final_schema,
      config_name,
      column_origin,
      log_name,
      concurrent_request,
      download_delay,
      enabledProxy,
      AUTOTHROTTLE_ENABLED,
      RANDOM_UA_PER_PROXY,
      AUTOTHROTTLE_DEBUG,
      ROBOTSTXT_OBEY,
      PROXYMESH_TIMEOUT,
      DEPTH_PRIORITY,
      AUTOTHROTTLE_TARGET_CONCURRENCY,
      AUTOTHROTTLE_START_DELAY,
      middlewareList
    }
  });
}

function getCounties(stateName){
  return apiClientAssessor.get(`${serverAddr}/assessor/available/us/${stateName}`).then(res => {
    if(Array.isArray( res.data)){
        return {data:res.data,is_array:true};
    }

    return {data:Object.keys(res.data), is_array:false};
  });
}

function getConfigsAssesor(stateName,configName){
  return apiClientAssessor.get(`${serverAddr}/assessor/available/us/${stateName}/${configName}`).then(res => {

    return res.data;
  });
}

function getCountries() {
    return apiClientAssessor.get(`${serverAddr}/assessor/available/`).then(res => {

    return Object.keys(res.data);
  });
}

function getStates(state = 'us'){
  return apiClientAssessor.get(`${serverAddr}/assessor/available/${state}`).then(res => {

    if(Array.isArray( res.data)){
        return {data:res.data,is_array:true};
    }

    return {data:Object.keys(res.data), is_array:false};
  });
}





function getDataOriginFields(dataOriginName){
    return apiClientAssessor.post(`${serverAddr}/assessor/data_origins/columns`,
        {data_origin_table:dataOriginName}).then(res => {
            return res.data;
          });
}

function getConfigs(name) {
  return apiClientAssessor.get("/connectors/configs", {
    params: { name }
  });
}

export {
  clientsAssessor,
  deleteClient,
  updateClient,
  getClient,
  runSpider,
  getConfigs,
  getDataOrigins,
  getDataOriginFields,
    getAllClientData,
  getCounties,
    getStates,
    getConfigsAssesor,
    newAssessorClient,
    getIntervals,
    getTaskType,
    newAssessorTask,
    getTaskResult,
    getTasks,
    getTaskResultData,
    newAssessorDataOrigin,
    getNextRunList,
    getSchemas,
    getSchema,getCountries,
    getTaskResultDataFamily,
    newConsolidationSchema
};
