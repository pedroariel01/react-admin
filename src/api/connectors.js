import { apiClient } from "./httpClient";

function getConnector(connector_id) {
  return apiClient.get(`/connectors/connector/${connector_id}`);
}

function getConnectors(client_id, page = 0) {
  let pageNumb = 1;
  if(page){
    const aux  = page.match('page=([0-9]+)');
    if(aux){
      pageNumb=aux[1]
    }
  }

  if (client_id) {
    return apiClient({
      url: `/connectors/all?client_id=${client_id}&page=${pageNumb}`,
      method: "GET"
    });
  } else {
    return apiClient({
      url: `/connectors/all?page=${pageNumb}`,
      method: "GET"
    });
  }
}

function getAssociationClient(client_id) {
  if (client_id) {
    return apiClient({
      url: `/connectors/client/association?search=${client_id}`,
      method: "GET"
    });
  }

  return apiClient({
    url: "/connectors/client/association",
    method: "GET"
  });
}

function buildAssociation(client_id, connector_id, filter) {
  return apiClient.post("/connectors/association/connector", {
    client_id,
    connector_id,
    filter
  });
}

function filterConnectors() {}

function getStatusConnector() {}

function updateStatusConnector() {}

export {
  getConnector,
  filterConnectors,
  getStatusConnector,
  updateStatusConnector,
  getConnectors,
  buildAssociation,
  getAssociationClient
};
