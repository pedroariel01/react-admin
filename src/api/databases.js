import { apiClient } from "./httpClient";

function database_list() {
  return apiClient.get("/connectors/databases");
}

function databaseCredentialList() {
  return apiClient.get("/connectors/database/credentials").then(res => {
    return res.data;
  });
}

function databasesFromHost(databaseCredId) {
  return apiClient({
    url: "/connectors/databasebyHost",
    method: "GET",
    params: { search: databaseCredId }
  }).then(res => {
    return res.data;
  });
}

function schemasByDatabase(databaseCredId, databasename) {
  return apiClient({
    url: "/connectors/schemasByDatabase",
    method: "GET",
    params: { databaseCredId, databasename }
  });
}

function newDatabaseCredential(dataCredential) {
  const data = {
    host: dataCredential.host.value,
    username: dataCredential.username.value,
    password: dataCredential.password.value
  };

  return apiClient.post("/connectors/credential/new", data).then(res => {
    return res;
  });
}

function updateDatabaseCredential(dataCredential) {
  const data = {
    id: dataCredential.id,
    host: dataCredential.host.value,
    username: dataCredential.username.value,
    password: dataCredential.password.value
  };

  return apiClient.post("/connectors/credential/update", data).then(res => {
    return res;
  });
}

function deleteCredential(id) {
  const data = {
    id
  };

  return apiClient.post("/connectors/credential/delete", data).then(res => {
    return res;
  });
}

function tablesBySchema(databaseCredId, databasename, schemaName) {
  return apiClient({
    url: "/connectors/tablesBySchema",
    method: "GET",
    params: { databaseCredId, databasename, schemaName }
  });
}

export {
  database_list,
  databaseCredentialList,
  newDatabaseCredential,
  updateDatabaseCredential,
  deleteCredential,
  databasesFromHost,
  schemasByDatabase,
  tablesBySchema
};
