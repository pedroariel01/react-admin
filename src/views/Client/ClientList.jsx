import React, {Component} from "react";
import {Button, Col, Grid, Row, Table} from "react-bootstrap";

import Card from "../../components/Card/Card.jsx";
import {Link, NavLink} from "react-router-dom";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
import Loader from "react-loader-spinner";
import {confirmAlert} from "react-confirm-alert"; // Import
import "react-confirm-alert/src/react-confirm-alert.css"; // Import css
import {clients, deleteClient} from "../../api/clients";
import Pager from "react-bootstrap/lib/Pager";

class ClientList extends Component {
    state = {
        clientList: [],
        next_page : '',
        prev_page : '',
        total:0,
        loading: true
    };

    componentDidMount() {
        clients().then(data => {
             console.log(data);
            this.setState({clientList: data.results, loading: false,
            total:data.count, next_page:data.next, prev_page:data.previous});
        });
    }

    delete = client => {
        const {name} = client;

        confirmAlert({
            title: "Confirm to delete",
            message: `Are you sure to delete client ${name}`,
            buttons: [
                {
                    label: "Yes",
                    onClick: () => {
                        this.setState({loading: true});

                        deleteClient(client).then(res => {
                            this.setState({clientList: res.data, loading: false});
                        });
                    }
                },
                {label: "No"}
            ]
        });
    };

    render() {
        const {clientList, loading} = this.state;

        const clientMarkup = clientList.map(client => {
            return (
                <tr key={client.id}>
                    <td>{client.name}</td>
                    <td>{client.association.length}</td>
                    <td>{client.data_count.permits}</td>
                    <td className="table-action-client">
                        <Link
                            title="Detail client"
                            key={client.id}
                            to={{
                                pathname: "/admin/clientconnectors/" + client.id,
                                state: {
                                    client: client
                                }
                            }}
                        >
                            <i className="mr-5 pe-7s-search action-icon"/>
                        </Link>

                        <Link to={"client/" + client.id} title="Edit client">
                            <i className="pe-7s-note mr-5 edit-icon action-icon"/>
                        </Link>

                        <span onClick={_ => this.delete(client)} title="Delete client">
              <i className="pe-7s-trash trash-icon action-icon"/>
            </span>
                    </td>
                </tr>
            );
        });

        return (
            <>
                <h3 className="title-page">Client List</h3>
                <div className="content">
                    <Grid fluid>
                        <Row>
                            <Col md={12}>
                                <NavLink to="/admin/clients/new" className="nav-link">
                                    <Button className={"btn-fill"} bsStyle="success">
                                        New Client
                                    </Button>
                                </NavLink>
                                {loading && (
                                    <Loader
                                        type="Circles"
                                        color="rgb(161, 232, 44)"
                                        height={100}
                                        width={100}
                                        className="loader"
                                    />
                                )}
                                {clientList.length > 0 && !loading && (
                                    <Card
                                        ctTableFullWidth
                                        ctTableResponsive
                                        content={
                                            <Table striped hover>
                                                <thead>
                                                <tr>
                                                    <td>Name</td>
                                                    <td>Connectors count</td>
                                                    <td>Permits</td>
                                                    <td>Action</td>
                                                </tr>
                                                </thead>
                                                <tbody>{clientMarkup}</tbody>
                                                <tfoot>
                                                <Pager>
                                                      <Pager.Item previous onClick={(event) => this.pagePrev(event)}>
                                                        &larr; Previous Page
                                                      </Pager.Item>
                                                      <Pager.Item next onClick={(event) => this.pageNext(event)}>
                                                        Next Page &rarr;
                                                      </Pager.Item>
                                                    </Pager>
                                                </tfoot>
                                            </Table>
                                        }
                                    />
                                )}
                            </Col>
                        </Row>
                    </Grid>
                </div>
            </>
        );
    }

    pageNext = e => {
        const {next_page} = this.state;
        if(next_page){
            clients(next_page).then(data => {
             console.log(data);
            this.setState({clientList: data.results, loading: false,
            total:data.count, next_page:data.next, prev_page:data.previous});
        });
        }

    };

    pagePrev = e => {
        const {prev_page} = this.state;
        if(prev_page){
            clients(prev_page).then(data => {
             console.log(data);
            this.setState({clientList: data.results, loading: false,
            total:data.count, next_page:data.next, prev_page:data.previous});
        });
        }

    };
}

export default ClientList;
