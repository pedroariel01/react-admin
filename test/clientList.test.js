import ClientList from "../src/views/Client/ClientList";

import React from "react";

import { render, cleanup, screen, waitFor } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

import { clients } from "../src/api/clients";

jest.mock("../src/api/clients");

const clientsDataMock = [
  {
    id: 1,
    association: [1, 2, 3, 4, 48],
    name: "SEATTLE",
    client_url: null,
    confluence_pages_urls: "",
    description: "Seattle Client",
    created_on: "2020-04-24T21:07:41.540798Z",
    notes: null
  },
  {
    id: 123,
    association: [40],
    name: "LONGVIEW",
    client_url: null,
    confluence_pages_urls: null,
    description: null,
    created_on: "2020-04-27T00:00:00Z",
    notes: null
  }
];

it("<ClientList/> renders data", async () => {
  clients.mockResolvedValueOnce(clientsDataMock);

  // render(
  //   <Router>
  //     <ClientList />
  //   </Router>
  // );
  const { asFragment } = render(
    <Router>
      <ClientList />
    </Router>
  );

  expect(clients).toHaveBeenCalledTimes(1);
  expect(clients).toHaveBeenCalledWith();
  await waitFor(() =>
    expect(screen.getByText("Connectors count")).toBeInTheDocument()
  );
  clientsDataMock.forEach(client => {
    expect(screen.getByText(client.name)).toBeInTheDocument();
  });

  expect(
    asFragment(
      <Router>
        <ClientList />
      </Router>
    )
  ).toMatchSnapshot();

});
