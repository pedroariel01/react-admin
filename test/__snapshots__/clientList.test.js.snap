// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClientList/> renders data 1`] = `
<DocumentFragment>
  <h3
    class="title-page"
  >
    Client List
  </h3>
  <div
    class="content"
  >
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <a
            class="nav-link"
            href="/admin/clients/new"
          >
            <button
              class="btn-fill btn btn-success"
              type="button"
            >
              New Client
            </button>
          </a>
          <div
            class="card"
          >
            <div
              class="header"
            >
              <h4
                class="title"
              />
              <p
                class="category"
              />
            </div>
            <div
              class="content table-full-width table-responsive"
            >
              <table
                class="table table-striped table-hover"
              >
                <thead>
                  <tr>
                    <td>
                      Name
                    </td>
                    <td>
                      Connectors count
                    </td>
                    <td>
                      Permits
                    </td>
                    <td>
                      Action
                    </td>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      SEATTLE
                    </td>
                    <td>
                      5
                    </td>
                    <td>
                      permit_amount
                    </td>
                    <td
                      class="table-action-client"
                    >
                      <a
                        href="/admin/clientconnectors/1"
                        title="Detail client"
                      >
                        <i
                          class="mr-5 pe-7s-search action-icon"
                        />
                      </a>
                      <a
                        href="/client/1"
                        title="Edit client"
                      >
                        <i
                          class="pe-7s-note mr-5 edit-icon action-icon"
                        />
                      </a>
                      <span
                        title="Delete client"
                      >
                        <i
                          class="pe-7s-trash trash-icon action-icon"
                        />
                      </span>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      LONGVIEW
                    </td>
                    <td>
                      1
                    </td>
                    <td>
                      permit_amount
                    </td>
                    <td
                      class="table-action-client"
                    >
                      <a
                        href="/admin/clientconnectors/123"
                        title="Detail client"
                      >
                        <i
                          class="mr-5 pe-7s-search action-icon"
                        />
                      </a>
                      <a
                        href="/client/123"
                        title="Edit client"
                      >
                        <i
                          class="pe-7s-note mr-5 edit-icon action-icon"
                        />
                      </a>
                      <span
                        title="Delete client"
                      >
                        <i
                          class="pe-7s-trash trash-icon action-icon"
                        />
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
              <div
                class="footer"
              >
                <div
                  class="stats"
                >
                  <i />
                   
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
